evml/keras/models.py

```diff
@@ -195,79 +267,251 @@ def load_model(cls, conf):
                 f"No saved model exists at {weights}. You must train a model first. Exiting."
             )
 
-        logger.info(
-            f"Loading a RegressorDNN with pre-trained weights from path {weights}"
+        logging.info(
+            f"Loading a DNN with pre-trained weights from path {weights}"
         )
         model_class = cls(**conf["model"])
         model_class.build_neural_network(
             len(conf["data"]["input_cols"]), len(conf["data"]["output_cols"])
         )
         model_class.model.load_weights(weights)
         return model_class
-
+    
+    def mae(self, y_true, y_pred):
+        num_splits = y_pred.shape[-1]
+        if num_splits == 4:
+            mu, _, _, _ = tf.split(y_pred, num_splits, axis=-1) 
+        elif num_splits == 2:
+            mu, _ = tf.split(y_pred, num_splits, axis=-1)
+        else:
+            mu = y_pred  # Assuming num_splits is 1
+        return tf.keras.metrics.mean_absolute_error(y_true, mu)
+    
+    def mse(self, y_true, y_pred):
+        num_splits = y_pred.shape[-1]
+        if num_splits == 4:
+            mu, _, _, _ = tf.split(y_pred, num_splits, axis=-1)
+        elif num_splits == 2:
+            mu, _ = tf.split(y_pred, num_splits, axis=-1)
+        else:
+            mu = y_pred  # Assuming num_splits is 1
+        
+        return tf.keras.metrics.mean_squared_error(y_true, mu)
+    
     def predict(self, x, scaler=None, batch_size=None):
+        """
+        Predict target values for input data.
+
+        Args:
+            x (numpy.ndarray): Input data.
+            scaler (optional): Scaler object for preprocessing input data (default: None).
+            batch_size (optional): Batch size for prediction (default: None).
+            y_scaler (optional): Scaler object for post-processing predicted target values (default: None).
+
+        Returns:
+            numpy.ndarray: Predicted target values.
+        """
         _batch_size = self.batch_size if batch_size is None else batch_size
         y_out = self.model.predict(x, batch_size=_batch_size)
+        if scaler:
+            if len(y_out.shape) == 1:
+                y_out = np.expand_dims(y_out, 1)
+            y_out = scaler.inverse_transform(y_out)
         return y_out
+    
+    def predict_ensemble(self, x, weight_locations, batch_size=None, scaler=None, num_outputs=1):
+        num_models = len(weight_locations)
```

Comment:
Can you add a docstring for predict_ensemble? What is weight_locations supposed to be?

Reply:
Done! Its a list of paths to saved weights.